SECURITY.md

```diff
@@ -0,0 +1,79 @@
+# Extensions Security Guidelines
+
+OpenSearch's support for extensions allows for taking already powerful use cases and expanding on them. With this increased functionality comes a larger surface area for misuse, vulnerabilities, and malicious interactions.
+
+By capturing the current state of OpenSearch ecosystem  and the plans for extensions this document outlines several areas for enhancements, features, and practices to incorporate into extensions for OpenSearch.
+
+## Document Phases
+These guidelines and this document are meant to evolve, the follow list captures the different phases this document will undergo.  Some areas might be complete ahead of others. Some areas or items might be marked as invalid/removed using markdown's strike-through.
+
+1. [X] Agreement of areas and 'as-is' state of OpenSearch Plugins and Extensions. **<-- Doc is here**
+2. [ ] All areas have recommendations and areas of investigation are filed as issues and linked back on this document.
+3. [ ] All investigation conclusions are captured and linked in this document, effectively define the scope of work for these areas.  Implementation of work is can be completed or outstanding.
+4. [ ] All planned work has been completed, issues around this work can be completed or outstanding. 
+5. [ ] Document complete, future work and issue will be captured out of band instead of as updates this document.
+
+## Terms
+To keep concepts consistent, this document is using terminology from [NIST Glossary](https://csrc.nist.gov/glossary).
+
+Additional terms:
+* **Plugin** - reference to the existing functionality to extend OpenSearch functionality.
+* **Extension** - reference to the in development functionality to extend OpenSearch.
+
+# Areas
+
+## Host security
+
+Plugins depend on use of the Java Security Manager is use to limit interactions on the host operation system resources (cpu/disk/memory/network/...).  JSM has been deprecated, with its removal scheduled in the next release of the JVM, see [OpenSearch discussion](https://github.com/opensearch-project/OpenSearch/issues/1687). Additional measures are needed to protect system resources.
+
+Extensions are sandboxed from the host system by operating via REST APIs.  This security boundary isolates extensions from executing operation system calls directly on OpenSearch hosts.
```

Comment:
This is only true for out-of-process extensions. How about the ones that can run in-process? I think we are still open to having those supported down the line

Reply:
AFAIK in process extensions are not being designed or implemented at this time.  How do you feel about not including it until this effort is more concrete?

Reply:
I think defining "REST API" is too specific for the potential application. The key point to make here is that interaction is limited to a defined set of API locations.  For external extensions, they will use REST.  If REST is not needed (e.g., in-process) there still should be another method of using the same API point.

Reply:
If extensions are available in-process, it seems like we are recreating the existing plugin feature with its same architectural weaknesses.   With a solid security boundary between OpenSearch and its extensions, we can build extensibility in alignment with trustworthy functionality.  I think there is a discussion to be had for in-process extensions and I would prefer to wait to add those considerations to this document until we are committing more thought in this space, does this seem like a worthwhile trade-off in the near term?

I am only away of using REST APIs for extensions, if we want to broaden the language do you have a suggestion?

Reply:
Looked into this further and I'm fine with leaving this as REST API for simplicity.